wpeverest/user-registration
# UR-1657 Enhance - Display notice about possible spam registrations if source different from our plugin's form

### All Submissions:






### Changes proposed in this Pull Request:
We have got a lot of complaints from users about spam users being registered on their sites. We have provided a lot of methods like recaptcha v2, v3, Honeypot, Cloudflare Turnstile, Custom Captcha Field, and Akismet Integration. Regardless of these spam users are being registered. Two cases can be here i.e. Users are not being able to set up Captcha or spam protection mechanisms or Users are being registered from sources other than our Registration Form.

We have documentation for the former one. This PR tries to provide a hint to the admin on the later one. Since users may be registered from sources other than our plugin's form, a user counting system has been added that counts a number of users that are registered from different sources in the past hour. If it exceeds 20 then a notice will be displayed to the admin. 

### How to test the changes in this Pull Request:

1. In a timeframe of 1 hour, register at least 20 users from other sources like, Add users from Users -> Add New, wp-admin's registration plugin or some other registration plugin's form.
2. See if a notice is being displayed to the admin dashboard if the user count exceeds 20, and check if links or buttons in the notice are being displayed properly.

### Types of changes:




### Other information:




### Changelog entry

> Enhance - Display notice about possible spam registrations if source different from our plugin's form.
